### PR TITLE
Call Slice in GetMemory

### PR DESCRIPTION
**Pull request by jkotalik**
For #7031

### CODE REVIEW

**Review by Tratcher — APPROVED** [review]

File: src/Http/Http/test/StreamPipeWriterTests.cs [hunk 1]
```diff
@@ -316,6 +316,30 @@ public async Task UseStreamThenPipeToWriteMultipleTimes()
             Assert.Equal(expectedString, ReadAsString());
         }
 
+        [Fact]
+        public void GetMemorySlicesCorrectly()
+        {
+            var expectedString = "abcdef";
+            var memory = Writer.GetMemory();
+
+            Encoding.ASCII.GetBytes(expectedString).CopyTo(memory);
```
Comment (pakrym) [inline src/Http/Http/test/StreamPipeWriterTests.cs]: I'm not a fan of these unit tests, in my opinion they are testing the wrong thing. 

Why not check that correct data is written to the stream when calling GetMemory() - Advance() multiple times?

**Comment by jkotalik** [comment]
@pakrym updated.

**Comment by pakrym** [comment]
LGTM

**Review by davidfowl — APPROVED** [review]

**Comment by JamesNK** [comment]
:shipit:

**Pull request merged**